verboseTree-maven-plugin/src/main/java/org/apache/maven/dependency/graph/DependencyGraphBuilder.java

```diff
@@ -39,14 +75,242 @@
     @Parameter( defaultValue = "${session}", readonly = true, required = true )
     protected MavenSession session;
 
-    @Component
+    @Parameter( defaultValue = "${repositorySystemSession}" )
+    private RepositorySystemSession repositorySystemSession;
+
+    @Parameter
+    private String outputFile;
+
+    @Inject
+    private ProjectDependenciesResolver resolver;
+
+    @Inject
     private ArtifactHandlerManager artifactHandlerManager;
 
+    private SerializeGraph serializer;
+
+    private DependencyNode rootNode;
+
+    private static final RepositorySystem system = RepositoryUtility.newRepositorySystem();
+
+    /**
+     * Maven repositories to use when resolving dependencies.
+     */
+    private final List<RemoteRepository> repositories;
+    private Path localRepository;
+
+    public DependencyGraphBuilder()
+    {
+        this( Arrays.asList( CENTRAL.getUrl() ) );
+    }
+
+    static
+    {
+        for ( Map.Entry<String, String> entry : OsProperties.detectOsProperties().entrySet() )
+        {
+            System.setProperty( entry.getKey(), entry.getValue() );
+        }
+    }
+
+    /**
+     * @param mavenRepositoryUrls remote Maven repositories to search for dependencies
+     * @throws IllegalArgumentException if a URL is malformed or does not have an allowed scheme
+     */
+    public DependencyGraphBuilder( Iterable<String> mavenRepositoryUrls )
+    {
+        List<RemoteRepository> repositoryList = new ArrayList<RemoteRepository>();
+        for ( String mavenRepositoryUrl : mavenRepositoryUrls )
+        {
+            RemoteRepository repository = mavenRepositoryFromUrl( mavenRepositoryUrl );
+            repositoryList.add( repository );
+        }
+        this.repositories = repositoryList;
+    }
+
     public void execute() throws MojoExecutionException
     {
-        getLog().info( project.getArtifactId() );
+        File file = new File( project.getBasedir().getAbsolutePath().replace( '\\', '/' ) + "/target/tree.txt" );
+
+        List<org.apache.maven.model.Dependency> dependencies = project.getDependencies();
+
+        rootNode = buildFullDependencyGraph( dependencies );
```

Comment:
Why is this a field? Looks like it could be a local variable.

Reply:
In the maven-dependency-plugin version of this code it's a field because there's a method to return the graph so I left it like that. I can change it though.